src/cow-react/common/containers/CancelButton/index.tsx

```diff
@@ -0,0 +1,59 @@
+import { useCallback, useEffect, useState } from 'react'
+import { CancelButtonContent, CancelButtonType } from '@cow/common/pure/CancelButton'
+import { useCancelOrder } from '@cow/common/hooks/useCancelOrder'
+import { SupportedChainId } from 'constants/chains'
+
+export type CancelButtonProps = {
+  chainId: SupportedChainId
+  summary: string | undefined
```

Comment:
why not optional instead of `| undefined`?

Reply:
Because it's a prop coming from parent which will always be set, but the value might be undefined.